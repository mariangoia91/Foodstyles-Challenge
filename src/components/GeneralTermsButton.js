import React from "react"
import { StyleSheet, View, Text, Alert } from "react-native"

function displayGeneralTerms() {
  Alert.alert(
    "General Terms",
    "These are the general terms and conditions",
    [{ text: "OK", onPress: () => console.log("Exited General Terms") }],
    { cancelable: false }
  )
}

const GeneralTermsButton = props => {
  return (
    <View style={{ bottom: 32, position: "absolute" }}>
      <Text style={styles.term_service}>
        By signing in I accept the{" "}
        <Text
          onPress={() => displayGeneralTerms()}
          style={(styles.term_service, { textDecorationLine: "underline" })}
        >
          General Terms
        </Text>
      </Text>
    </View>
  )
}

const styles = StyleSheet.create({
  term_service: {
    fontFamily: "Montserrat-Regular",
    textAlign: "center",
    fontWeight: "500",
    fontSize: 16,
    color: "white"
  }
})

export default GeneralTermsButton
