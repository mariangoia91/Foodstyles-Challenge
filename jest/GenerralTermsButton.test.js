import React from "react"
import GeneralTermsButton from "../src/components/GeneralTermsButton"

import renderer from "react-test-renderer"

test("renders correctly", () => {
  const tree = renderer.create(<GeneralTermsButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
