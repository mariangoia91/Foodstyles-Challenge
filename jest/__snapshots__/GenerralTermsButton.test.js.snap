// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 32,
      "position": "absolute",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Montserrat-Regular",
        "fontSize": 16,
        "fontWeight": "500",
        "textAlign": "center",
      }
    }
  >
    By signing in I accept the
     
    <Text
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      General Terms
    </Text>
  </Text>
</View>
`;
